canonicalwebteam/blog/wordpress_api.py

```diff
@@ -83,7 +83,7 @@ def get_articles_with_metadata(**kwargs):
 
     :returns: response, metadata dictionary
     """
-    url = build_get_article_url(**kwargs)
+    url = build_get_articles_url(**kwargs)
```

Comment:
Related to what we are discussion about testing, I think you should switch around these interfaces. `get_articles` and `get_articles_with_metadata` should take *explicit* arguments for the functions they expect to fulfill - `page` etc.

Then, I think it would be reasonable to change `build_get_articles_url` to be flexible to its parameters, as it doesn't actually need any specific knowledge - all it does is:

- Convert arrays into comma-separated strings
- Convert booleans into `name={bool}` strings
- Compile the query string

So I actually think you could happily genericise `build_get_articles_url` into a `_build_query` function, which takes a `parameters` argument (I consider `**kwargs` to be a bit of a potentially-confusing hack that we should avoid if we can), and then does the transformations I describe above.

Again, I don't think this function itself should need testing, its usage by `get_articles` and every other extenal function that makes an API call with a query string (whose *entire* interface should be tested) should provide quite enough of a test.

Reply:
Yup, @jpmartinspt mentioned this before. I think Ill ask him for some help on the urllib and reopen that issue

Reply:
@nottrobin this is the issue https://github.com/canonical-web-and-design/canonicalwebteam.blog/issues/15 .

Id rather not include this into this PR as well :)